src/target/rp.c

```diff
@@ -617,26 +641,27 @@ static bool rp_cmd_erase_sector(target *t, int argc, const char **argv)
 	} else if (argc == 2)
 		length = strtoul(argv[1], NULL, 0);
 	else
-		return -1;
+		return false;
 
 	rp_priv_s *ps = (rp_priv_s *)t->target_storage;
 	ps->is_monitor = true;
-	const bool result = rp_flash_erase(t->flash, start, length);
+	bool ret = true; /* catch false returns with &= */
```

Comment:
keeping the "result" naming here would be good to be honest, 'ret' in other places in the code is something we've let in but we're not especially happy with

Reply:
Fixed!